src/tests/ftest/control/daos_snapshot.py

```diff
@@ -0,0 +1,142 @@
+        super(DaosSnapshotTest, self).__init__(*args, **kwargs)
+        self.daos_cmd = None
+
+    def create_snapshot(self, pool_uuid, svc, cont_uuid, count):
+        """Create snapshots and return the epoch values obtained from stdout.
+
+        Args:
+            pool_uuid (String): Pool UUID.
+            svc (String): Service replicas.
+            cont_uuid (String): Container UUID.
+            count (Integer): Number of snapshots to create.
+
+        Returns:
+            List of string: Epochs obtained from stdout.
+        """
+        kwargs = {"pool": pool_uuid, "svc": svc, "cont": cont_uuid}
+        epochs = []
+        for _ in range(count):
+            epochs.append(
+                self.daos_cmd.get_output("container_create_snap", **kwargs)[0])
+        return epochs
+
+    def test_create_list_delete(self):
+        """JIRA ID: DAOS-4872
+
+        Test Description:
+            Test daos container create-snap, list-snaps, destroy-snap
+
+        Use Cases:
+            See test cases in the class description.
+
+        :avocado: tags=all,small,control,full_regression,daos_snapshot
+        """
+        kwargs = {"scm_size": "1G"}
+        data = self.get_dmg_command().pool_create(scm_size="1G")
+        pool_uuid = data["uuid"]
+        sr = data["svc"]
+        self.daos_cmd = DaosCommand(self.bin)
+        kwargs = {"pool": pool_uuid, "svc": sr}
+        cont_uuid = self.daos_cmd.get_output("container_create", **kwargs)[0]
+
+        # 1. Create 5 snapshots.
+        expected_epochs = self.create_snapshot(
+            pool_uuid=pool_uuid, svc=sr, cont_uuid=cont_uuid, count=5)
+        expected_epochs.sort()
+        self.log.debug("Expected Epochs = {}".format(expected_epochs))
+        kwargs = {"pool": pool_uuid, "svc": sr, "cont": cont_uuid}
+
+        # 2. List the 5 snapshots and verify their epochs.
+        actual_epochs = self.daos_cmd.get_output(
+            "container_list_snaps", **kwargs)
+        actual_epochs.sort()
+        self.log.debug("Actual Epochs = {}".format(actual_epochs))
+        self.assertEqual(expected_epochs, actual_epochs)
+
+        # 3. Destroy all the snapshots.
+        for epoch in actual_epochs:
+            _ = self.daos_cmd.container_destroy_snap(
+                pool=pool_uuid, cont=cont_uuid, epochs=[epoch], svc=sr)
+
+        # 4. List and verify that there's no snapshot.
+        epochs = self.daos_cmd.get_output("container_list_snaps", **kwargs)
+        self.assertTrue(not epochs)
+
+    @skipForTicket("DAOS-4691")
+    def test_epcrange(self):
+        """JIRA ID: DAOS-4872
+
+        Test Description:
+            Test --epcrange. See class description.
+
+        Use Cases:
+            See class description.
+        
+        :avocado: tags=all,small,container,full_regression,daos_snapshot_range
+        """
+        # 1. Create 5 snapshots.
+        new_epochs = self.create_snapshot(
+            pool_uuid=pool_uuid, svc=sr, cont_uuid=cont_uuid, count=5)
```

Comment:
This code does not look complete:  `pool_uuid`, `sr`, and `cont_uuid` are undefined.  Even though the test is skipped, it will create a problem for whomever comes along and removes the skip later on.

Reply:
Done